src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -196,6 +286,256 @@ IFACEMETHODIMP_(void) ZoneSet::MoveSizeEnd(HWND window, HWND zoneWindow, POINT p
     }
 }
 
+IFACEMETHODIMP_(void)
+ZoneSet::CalculateZones(MONITORINFO monitorInfo, JSONHelpers::ZoneSetLayoutType layoutType, int zoneCount, int spacing, const std::wstring& customZoneSetFilePath) noexcept
+{
+    Rect const workArea(monitorInfo.rcWork);
+
+    switch (layoutType)
+    {
+    case JSONHelpers::ZoneSetLayoutType::Focus:
+        CalculateFocusLayout(workArea, zoneCount);
+        break;
+    case JSONHelpers::ZoneSetLayoutType::Columns:
+    case JSONHelpers::ZoneSetLayoutType::Rows:
+        CalculateColumnsAndRowsLayout(workArea, layoutType, zoneCount, spacing);
+        break;
+    case JSONHelpers::ZoneSetLayoutType::Grid:
+    case JSONHelpers::ZoneSetLayoutType::PriorityGrid:
+        CalculateGridLayout(workArea, layoutType, zoneCount, spacing);
+        break;
+    case JSONHelpers::ZoneSetLayoutType::Custom:
+        CalculateCustomLayout(workArea, customZoneSetFilePath, spacing);
+        break;
+    }
+}
+
+void ZoneSet::CalculateFocusLayout(Rect workArea, int zoneCount) noexcept
+{
+    LONG left{ static_cast<LONG>(workArea.width() * 0.1) };
+    LONG top{ static_cast<LONG>(workArea.height() * 0.1) };
+    LONG right{ static_cast<LONG>(workArea.width() * 0.6) };
+    LONG bottom{ static_cast<LONG>(workArea.height() * 0.6) };
+    RECT focusZoneRect{ left, top, right, bottom };
+    int focusRectXIncrement = (zoneCount <= 1) ? 0 : (int)(workArea.width() * 0.2) / (zoneCount - 1);
+    int focusRectYIncrement = (zoneCount <= 1) ? 0 : (int)(workArea.height() * 0.2) / (zoneCount - 1);
+
+    for (int i = 0; i < zoneCount; i++)
+    {
+        AddZone(MakeZone(focusZoneRect));
+        focusZoneRect.left += focusRectXIncrement;
+        focusZoneRect.right += focusRectXIncrement;
+        focusZoneRect.bottom += focusRectYIncrement;
+        focusZoneRect.top += focusRectYIncrement;
+    }
+}
+
+void ZoneSet::CalculateColumnsAndRowsLayout(Rect workArea, JSONHelpers::ZoneSetLayoutType type, int zoneCount, int spacing) noexcept
+{
+    int gutter = spacing;
+    int c_multiplier = 10000;
```

Comment:
`constexpr int C_MULTIPLIER = 10000;` from the begining of the file can be used ?